filebeat/tests/integration/translate_guid_test.go

```diff
@@ -0,0 +1,217 @@
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/go-ldap/ldap/v3"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/v7/libbeat/tests/integration"
+	"github.com/elastic/elastic-agent-autodiscover/docker"
+)
+
+const translateguidCfg = `
+filebeat.inputs:
+  - type: filestream
+    id: "test-translateguidCfg"
+    paths:
+      - %s
+
+queue.mem:
+  flush.min_events: 1
+  flush.timeout: 0.1s
+
+path.home: %s
+
+output.file:
+  path: ${path.home}
+  filename: "output-file"
+
+logging:
+  metrics:
+    enabled: false
+
+processors:
+  - add_fields:
+      fields: 
+        guid: '%s'
+  - translate_guid:
+      field: fields.guid
+      target_field: fields.common_name
+      ldap_address: 'ldap://localhost:1389'
+      ldap_base_dn: 'dc=example,dc=org'
+      ldap_bind_user: 'cn=admin,dc=example,dc=org'
+      ldap_bind_password: 'adminpassword'
+      ldap_guid_attribute: 'entryUUID'
+`
+
+func TestTranslateGUIDWithLDAP(t *testing.T) {
+	startOpenldapContainer(t)
+
+	var entryUUID string
+	require.Eventually(t, func() bool {
+		var err error
+		entryUUID, err = getLDAPUserEntryUUID()
+		if err != nil {
+			return false
+		}
+		return true
+	}, 10*time.Second, time.Second)
+
+	filebeat := integration.NewBeat(
+		t,
+		"filebeat",
+		"../../filebeat.test",
+	)
+	tempDir := filebeat.TempDir()
+
+	// 1. Generate the log file path
+	logFilePath := path.Join(tempDir, "log.log")
+	integration.GenerateLogFile(t, logFilePath, 1, false)
+
+	// 2. Write configuration file and start Filebeat
+	filebeat.WriteConfigFile(
+		fmt.Sprintf(translateguidCfg, logFilePath, tempDir, entryUUID),
+	)
+	filebeat.Start()
+
+	var outputFile string
+	require.Eventually(t, func() bool {
+		outputFiles, err := filepath.Glob(path.Join(tempDir, "output-file-*.ndjson"))
+		if err != nil {
+			return false
+		}
+		if len(outputFiles) != 1 {
+			return false
+		}
+		outputFile = outputFiles[0]
+		return true
+	}, 10*time.Second, time.Second)
+
+	// 3. Wait for the event with the expected translated guid
+	filebeat.WaitFileContains(
+		outputFile,
+		fmt.Sprintf(`"fields":{"guid":"%s","common_name":["User1","user01"]}`, entryUUID),
+		10*time.Second,
+	)
+}
+
+func startOpenldapContainer(t *testing.T) {
+	ctx := context.Background()
+	c, err := docker.NewClient(client.DefaultDockerHost, nil, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	reader, err := c.ImagePull(ctx, "bitnami/openldap:2", image.PullOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	io.Copy(os.Stdout, reader)
+	reader.Close()
+
+	resp, err := c.ContainerCreate(ctx,
+		&container.Config{
+			Image: "bitnami/openldap:2",
```

Comment:
I'm not that familiar with LDAP, but here it is installing a 2.x openldap and the go client is v3. I wonder if these versions should match?

Reply:
The ldap library refers to the library version, openldap version is just the version of openldap server in this case. They are not correlated

Reply:
Got it, thanks!